feat: add encode kwargs to speedtasks

bndos opened the pull request:
This commit adds `encode_kwargs` passing to the `AbsTaskSpeedTask` class, adds `MockSpeedTask` for testing and reuses the `model_encode` function instead of `model.encode`.

This commit also ensures that the_ugly_duckling.txt file is present when installing `mteb[speedtask]`. Avoids running into a `FileNotFoundError` when trying to run speed tasks.

<!-- If you are submitting a dataset or a model for the model registry please use the corresponding checklists below otherwise feel free to remove them. -->

<!-- add additional description, question etc. related to the new dataset -->


## Checklist
<!-- Please do not delete this -->

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

dbuades committed:
fix: openai dimensions

clinia-brando committed:
fix: adds encode kwargs to speedtasks

dbuades commented:
Nice! I don't see any red flags. Let's discuss whether to move forward with this PR on Tuesday when discussing our benchmarking strategy.

dbuades committed:
feat: add text-multilingual-embedding-002

dbuades committed:
fix: bm25s search

x-tabdeveloping committed:
WIP: Polishing up leaderboard UI (#1461)

* fix: Removed column wrapping on the table, so that it remains readable

* Added disclaimer to figure

* fix: Added links to task info table, switched out license with metric

Samoed committed:
fix: loading pre 1.11.0 (#1460)

* small fix

* fix: fix

Commit:
1.19.8

Automatically generated by python-semantic-release

isaac-chung committed:
fix: swap touche2020 to maintain compatibility (#1469)

swap touche2020 for parity

Commit:
1.19.9

Automatically generated by python-semantic-release

isaac-chung committed:
docs: Add sum per language for task counts (#1468)

* add sum per lang

* add sort by sum option

* make lint

Napuh committed:
fix: pinned datasets to <3.0.0 (#1470)

Commit:
1.19.10

Automatically generated by python-semantic-release

dbuades committed:
feat: add CUREv1 retrieval dataset (#1459)

* feat: add CUREv1 dataset

---------

Co-authored-by: nadshe <nadia.sheikh@clinia.com>
Co-authored-by: olivierr42 <olivier.rousseau@clinia.com>
Co-authored-by: Daniel Buades Marcos <daniel@buad.es>

* feat: add missing domains to medical tasks

* feat: modify benchmark tasks

* chore: benchmark naming

---------

Co-authored-by: nadshe <nadia.sheikh@clinia.com>
Co-authored-by: olivierr42 <olivier.rousseau@clinia.com>

github-actions[bot] committed:
Update tasks table

Commit:
1.20.0

Automatically generated by python-semantic-release

dbuades committed:
feat: nvidia models

oliviercl42 committed:
fix: publichealthqa

Samoed committed:
fix: check if `model` attr of model exists (#1499)

* check if model attr of model exists

* lint

* Fix retrieval evaluator

Commit:
1.20.1

Automatically generated by python-semantic-release

x-tabdeveloping committed:
fix: Leaderboard demo data loading (#1507)

* Made get_scores error tolerant

* Added join_revisions, made get_scores failsafe

* Fetching metadata fixed fr HF models

* Added failsafe metadata fetching to leaderboard code

* Added revision joining to leaderboard app

* fix

* Only show models that have metadata, when filter_models is called

* Ran linting

Commit:
1.20.2

Automatically generated by python-semantic-release

x-tabdeveloping committed:
fix: leaderboard only shows models that have ModelMeta (#1508)

Filtering for models that have metadata

Commit:
1.20.3

Automatically generated by python-semantic-release

Samoed committed:
fix: align readme with current mteb (#1493)

* align readme with current mteb

* align with mieb branch

* fix test

Commit:
1.20.4

Automatically generated by python-semantic-release

isaac-chung committed:
docs: Add lang family mapping and map to task table (#1486)

* add lang family mapping and map to task table

* make lint

* add back some unclassified lang codes

github-actions[bot] committed:
Update tasks table

KennethEnevoldsen committed:
fix: Ensure that models match the names on embedding-benchmarks/results (#1519)

Commit:
1.20.5

Automatically generated by python-semantic-release

x-tabdeveloping committed:
fix: Adding missing metadata on models and mathcing names up with the results repo (#1528)

* Added Voyage 3 models

* Added correct metadata to Cohere models and matched names with the results repo

Commit:
1.20.6

Automatically generated by python-semantic-release

isaac-chung committed:
feat: Evaluate missing splits (#1525)

* fix: evaluate missing splits (#1268)

* implement partial evaluation for missing splits

* lint

* requested changes done from scratch

* test for missing split evaluation added

* uncomment test

* lint

* avoid circular import

* use TaskResult

* skip tests for now

---------

Co-authored-by: Isaac Chung <chungisaac1217@gmail.com>

* got test_all_splits_evaluated passing

* tests passing

* address review comments

* make lint

* handle None cases for kg_co2_emissions

* use new results info

---------

Co-authored-by: Thivyanth <thivyanth2004@gmail.com>

Commit:
1.21.0

Automatically generated by python-semantic-release

isaac-chung committed:
fix: Correct typos superseeded -> superseded (#1532)

fix typo -> superseded

Commit:
1.21.1

Automatically generated by python-semantic-release

isaac-chung committed:
fix: Task load data error for SICK-BR-STS and XStance (#1534)

* fix task load data for two tasks

* correct dataset keys

Commit:
1.21.2

Automatically generated by python-semantic-release

x-tabdeveloping committed:
fix: Proprietary models now get correctly shown in leaderboard (#1530)

* Fixed showing proprietary models in leaderboard

* Added links to all OpenAI models

* Fixed table formatting issues

* Bumped Gradio version

Commit:
1.21.3

Automatically generated by python-semantic-release

isaac-chung committed:
docs: Add Model Meta parameters and metadata (#1536)

* add multi_qa_MiniLM_L6_cos_v1 model meta
* add all_mpnet_base_v2
* add parameters to model meta
* make lint
* add extra params to meta

isaac-chung committed:
fix: add more model meta (jina, e5) (#1537)

* add e5 model meta

* address review comments

Commit:
1.21.4

Automatically generated by python-semantic-release

KennethEnevoldsen committed:
Add cohere models (#1538)

* fix: bug cohere names

* format

KennethEnevoldsen committed:
fix: add nomic models (#1543)

#1515

KennethEnevoldsen committed:
fix: Added all-minilm-l12-v2 (#1542)

#1515

KennethEnevoldsen committed:
fix: Added arctic models (#1541)

#1515

yjoonjang committed:
fix: add sentence trimming to OpenAIWrapper (#1526)

* fix: add sentence trimming to OpenAIWrapper

* fix: import tiktoken library inside encode function

* fix: check tokenizer library installed and update ModelMeta to pass tokenizer_name

* fix: pass tokenizer_name, max_tokens to loader

* fix: make tokenizer_name None for default

* fix: delete changes for ModelMeta

* fix: fix revision to 2 for OpenAI models

* fix: add docstring for OpenAIWrapper

* fix: lint

* feat: add openai optional dependency set

* fix: add sleep for too many requests

* fix: add lint

* fix: delete evaluate file

Commit:
1.21.5

Automatically generated by python-semantic-release

x-tabdeveloping committed:
fix: Fixed metadata errors (#1547)

Commit:
1.21.6

Automatically generated by python-semantic-release

KennethEnevoldsen committed:
fix: remove curev1 from multlingual (#1552)

Seems like it was added here:
https://github.com/embeddings-benchmark/mteb/commit/1cc6c9e0fe62ca4e77708b641823fa1a121f048b

Commit:
1.21.7

Automatically generated by python-semantic-release

x-tabdeveloping committed:
fix: Add Model2vec (#1546)

* Added Model2Vec wrapper

* Added Model2vec models

* Added model2vec models to registry

* Added model2vec as a dependency

* Ran linting

* Update mteb/models/model2vec_models.py

Co-authored-by: Kenneth Enevoldsen <kennethcenevoldsen@gmail.com>

* Update mteb/models/model2vec_models.py

Co-authored-by: Kenneth Enevoldsen <kennethcenevoldsen@gmail.com>

* Added adapted_from and superseeded_by to model2vec models.

* Added missing import

* Moved pyproject.toml to optional dependencies

* Fixed typos

* Added import error and changed model to model_name

* Added Numpy to frameworks

* Added Numpy to frameworks

* Corrected false info on model2vec models

* Replaced np.inf with maxint

* Update mteb/models/model2vec_models.py

Co-authored-by: Isaac Chung <chungisaac1217@gmail.com>

* Added option to have infinite max tokens, added it to Model2vec

---------

Co-authored-by: Kenneth Enevoldsen <kennethcenevoldsen@gmail.com>
Co-authored-by: Isaac Chung <chungisaac1217@gmail.com>

x-tabdeveloping committed:
Made result loading more permissive, changed eval splits for HotPotQA and DBPedia (#1554)

* Removed train and dev from eval splits on HotpotQA

* Removed dev from eval splits on DBPedia

* Made task_results validation more permissive

* Readded exception in get_score

* Ran linting

Commit:
1.21.8

Automatically generated by python-semantic-release

dbuades committed:
feat: add new google models

dbuades committed:
fix: propietary models

dbuades commented:
Closing this PR since all the relevant commits have already been merged upstream into `https://github.com/embeddings-benchmark/mteb`.

We'll reopen a PR for `speedtask` if needed once we get back to using it, I've kept the related commit in a separate branch.
